node/src/sub_lib/node_addr.rs

```diff
@@ -165,6 +132,39 @@ impl FromStr for NodeAddr {
     }
 }
 
+impl NodeAddr {
+    pub const PORTS_SEPARATOR: &'static str = "/";
+
+    pub fn new(ip_addr: &IpAddr, ports: &[u16]) -> NodeAddr {
+        let mut ports = ports.to_owned();
+        ports.sort_unstable();
```

Comment:
An interesting name...

I bet it's a completely innocent function though. :)

Reply:
A sort is stable if it preserves the order of equal elements. For example, if you have Stanley and Denise, both of whom are 23 years old, and you sort them by age, and you get back Stanley and Denise, then the sort is stable. If you get back (or _might_ get back) Denise and Stanley, then the sort is unstable.

With some sort algorithms--apparently including the one Rust's library collections use--unstable sorts are cheaper than stable sorts; so you choose unstable sorts unless you have a special need for the stable kind.